### DISCUSSION TITLE
Variant exe output?

### DISCUSSION THREAD

**Discussion by Flithor** [dotnet/msbuild#8550] [Ideas] [answered]
MSBuild will only generate one exe file for one exe project by default.
Is it possible to have it output more exe, and each launched main program with specific command line arguments?
Just like run a bat launcher.
Example:
```
Main.exe => run as command "Main"
Main.Lite.exe => run as command "Main -lite"
Main.Pro.exe => run as command "Main -pro"
Setting.exe => run as command "Main -set"
```

**Comment by jrdodds** [comment, marked as answer]
@Flithor Does the behavior of your .exe change based on the program name (i.e.. `Environment.GetCommandLineArgs()[0]`)? You can easily add a custom target to your project that makes copies of (or links to) the built exe.

**Reply by Flithor** [reply]
Oh mY GoD!
Genius idea!!!!!!
